controllers/release/adapter.go

```diff
@@ -337,14 +348,50 @@ func (a *adapter) EnsureReleaseProcessingIsTracked() (controller.OperationResult
 	return controller.ContinueProcessing()
 }
 
-// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
-// Processing step are cleaned up once processing is finished.
-func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if !a.release.HasProcessingFinished() {
+// EnsureReleaseManagedProcessingIsTracked is an operation that will ensure that the Release Managed PipelineRun status
+// is tracked in the Release being processed.
+func (a *adapter) EnsureReleaseManagedProcessingIsTracked() (controller.OperationResult, error) {
+	if !a.release.IsManagedProcessing() || a.release.HasManagedProcessingFinished() {
 		return controller.ContinueProcessing()
 	}
 
-	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release, metadata.ManagedPipelineType)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+	if pipelineRun != nil {
+		err = a.registerManagedProcessingStatus(pipelineRun)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// EnsureReleaseProcessingTenantResourcesAreCleanedUp is an operation that will ensure that the resources created for the
+// Release Tenant Processing step are cleaned up once processing is finished.
+func (a *adapter) EnsureReleaseTenantProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
```

Comment:
All of them can go together. We should search all the pipelines for the release (we have a label for that) and delete them in the same function.